implementations/rust/ockam/ockam_command/src/util/parsers.rs

```diff
@@ -30,6 +31,12 @@ pub(crate) fn socket_addr_parser(input: &str) -> Result<SocketAddr> {
     }
 }
 
+/// Helper fn for parsing an identity from user input by using
+/// [`ockam_identity::IdentityIdentifier::decode_hex()`]
+pub(crate) fn identity_identifier_parser(input: &str) -> Result<IdentityIdentifier> {
+    Ok(IdentityIdentifier::from_hex(input))
```

Comment:
I see! Once this change is made, and the parser is utilized, how may I obtain the Identity from the IdentityIdentifier? The credential store command, for instance, needs the IdentityIdentifier, and obtains it by hex decoding the string, but later also needs the Identity, when it passes it to

```rust
CredentialConfig::new()
```

Also, after the modification, should further changes be made to the bats tests?

Many thanks for your help!

Reply:
You can get Identifier from an Identity, but not vice versa. Does that make sense?

Reply:
I see - I'd thought there was a one-to-one mapping between the two, and, from the Identifier, one might recover the Identity itself. Although, as the command is receiving the value for the first time, there would be nowhere to go to retrieve it, only create it - although I am still to get a reasonable grasp of the concepts. If it is not possible to obtain the Identity from the Identifier, then can the parser not be used for `credential store` and `credential verify`?

Two other things I only now realize are that I neglected to document the changes to the command, as I did not see that the format of the `--for` argument would change, and that perhaps, from that, calling this a refactor might be inappropriate. I apologize for the back-and-forth, but hope I can learn from the experience and better contribute for it.

Given this Identifier-Identity situation, what would be best, and where is the place for editing the corresponding documentation?

Many thanks!

Reply:
@adrianbenavides @SanjoDeundiak Good day! Just bumping as I am stuck - apologies.

Reply:
Thanks for the help, @adrianbenavides! I have made some modifications which I believe will make the bats tests pass as soon as one other thing is dealt with, which is an error when trying to use IdentityIdentifier::from_str. It is failing at this point, in

ockam/implementations/rust/ockam/ockam_identity/src/identity/identity_identifier.rs:92

```rust
impl TryFrom<&str> for IdentityIdentifier {
    type Error = Error;

    fn try_from(value: &str) -> Result<Self> {
        let value = value.trim();
        if value.starts_with(Self::PREFIX) {
            Ok(Self(value.to_string()))
        } else {
            Err(IdentityError::InvalidIdentityId.into())
        }
    }
}
```

The PREFIX is just the uppercase letter "P". What would the argument to `ockam credential issue --for <ARGUMENT>` have to look like to be accepted, but still be a plain identity identifier?

Reply:
This is where it gets tricky.
- You want to use `identifier=$($OCKAM identity show i1 --encoding hex)` when you want to use that variable in `credential issue/verify` commands
- You want to use `idt1=$($OCKAM identity show i1 --full --encoding hex)` when you want to use that variable in `credential store` commands

Reply:
Make sure to pull the latest changes. I've rebased the latest develop changes to fix a conflict.

Reply:
Good day, @adrianbenavides! I thank you once again for the patience and help.

I've just made changes that pass the bats tests. However, there is a failing Rust test (ckam_command enroll::oidc_service::tests::test_wait_for_authorization_code) whose cause for failure I do not know. How does the failing Rust test relate to my changes?

Reply:
We recently modified those tests because they were flaky on CI. I've squashed your commits and rebased on top of latest develop. I'll do a final pass, but looks like we are ready to merge!